File: tags/alpha-Lazarus_initial/skychart/scripts/TcpClient1.java
/*
 * Created on 9 sept. 2005
 *
 * example program that receive all messages from the program
 * and process the information from each click on the chart.
 * 
*/
package demo;

import net.api.skychart.client.CdcClient;

/**
 * @author Patrick Chevalley
 *
 */
public class TcpClient1 {

	public static void main(String[] args) {
		CdcClient cdc = new CdcClient();
		if (cdc.connected()) {
			String buf,tab,chart,ra,dec,type,desc ;
			String[] rep;
			while ((buf=cdc.readLine())!=null) {
				if (buf.compareTo(".")!=0){  
					rep=buf.split("\t",6);
					if ((rep.length==6)&&(rep[0].compareTo(">")==0)){
						chart=rep[1];
						ra=rep[2];
						dec=rep[3];
						type=rep[4];
						desc=rep[5];
						System.out.println("From "+chart+"  RA:"+ra+" DEC:"+dec+" Type:"+type);
						System.out.println("Desc: "+desc);
					} else {
						System.out.println(buf);
					}
				}				
			}
				
			System.out.println("Exit!");				
		}
						
	}


}

File: trunk/skychart/scripts/tcpclient1.pl
#!/usr/bin/perl
use IO::Socket;

#
# example program that receive all messages from the program
# and process the information from each click on the chart.
#

$host = "192.168.0.1";
$port = "3292";

connectCDC();

while (defined ($line = <$handle>))
{
	if (!($line =~/^\.\r\n$/))  #skip keepalive data
	{
		($h1,$chart,$h2,$ra,$dec,$type,$desc) = split(/\s+/,$line,7);
		if ($ra and ($h1 eq ">"))
			{
			print "From $chart  RA: $ra DEC: $dec Type: $type \n";
			print "$desc \n";
			}
		else { print $line };
	}
}


sub connectCDC {

# do the connection
$handle = IO::Socket::INET->new(Proto     => "tcp",
                                PeerAddr  => $host,
                                PeerPort  => $port)
          or die "cannot connect to Cartes du Ciel at $host port $port : $!";

$handle->autoflush(1);

print STDOUT "[Connected to $host:$port]\n";

# wait connection and get client chart name
  $line = <$handle>;
  print STDOUT $line;
  $line =~ /OK! id=(.*) chart=(.*)$/;
  $client = $1;
  $chart = $2;
  chop $chart;
  if ($client)
    {
     print STDOUT " We are connected as client $client , the active chart is $chart\n";
     print STDOUT " Close CDC or hit Ctrl+C to quit.\n";
    }
    else { die " We are not connected \n"};
}
